docs/rules/no-leaky-event-listeners.md

```diff
@@ -0,0 +1,47 @@
+## Prevent event listeners from leaking memory (no-leaky-event-listeners)
+
+It is extremely easy to leak event listeners. In most of the cases, not only the event handler leaks, but also all the object capture into closure. This rules prevents common patterns associated with event listener leaking.
+
+### Rule details
+
+Example of **incorrect** code:
+
+```js
+import { LightningElement } from 'lwc';
+
+export default class Test extends LightningElement {
+    connectedCallback() {
+        window.addEventListener('test', handleTest.bind(this));
+        //                              ^ Event listener will leak.
+    }
+
+    disconnectedCallback() {
+        window.removeEventListener('test', handleTest.bind(this));
+        //                                 ^ Event listener will leak.
+    }
+
+    handleTest() {}
+}
+```
+
+In the code above the event listener will leak because invoking `Function.prototype.bind()` on a method returns a new method with a bound `this` value. Since the component doesn't have a reference to the newly created function, the added event listener can't be removed later on. In this case, since the `this` value is trapped into the generated function, each newly created component will be kept in memory.
+
+Example of **correct** code:
+
+```js
+import { LightningElement } from 'lwc';
+
+export default class Test extends LightningElement {
+    connectedCallback() {
+        window.addEventListener('test', this.handleTest);
+    }
+
+    disconnectedCallback() {
+        window.removeEventListener('test', this.handleTest);
+    }
+
+    handleTest = () => {};
```

Comment:
Thoughts on adding a second correct example that uses an expando on the instance?  Eg
```
export default class Test extends LightningElement {
    connectedCallback() {
        this.handleTestBound = this.handleTest.bind(this);
        window.addEventListener('test', this.handleTestBound);
    }
    disconnectedCallback() {
        window.removeEventListener('test', this.handleTestBound);
        this.handleTestBound = undefined;
    }


```

Reply:
I would rather have a single valid code, to avoid introducing confusion when reading the doc. IMO, we should present a single blessed pattern to solve this issue. I don't have a strong opinion on whether to use the class field pattern or the expando.

Reply:
This example needs to align with the recommended pattern. You've just decided on the recommended pattern :)_

 @diervo  @Gr8Gatsby  @jbleyleSF  are you good with this? 

@jbleyleSF  https://developer.salesforce.com/docs/component-library/documentation/en/lwc/events_handling needs to be updated with the recommended pattern. FYI it currently shows an anti-pattern.

@Gr8Gatsby the Trailhead sample apps need to be updated to reflect this change

Reply:
@pmdartus @kevinv11n Doc review: https://git.soma.salesforce.com/ContentX/lwc/pull/929

Reply:
@pmdartus @kevinv11n Doc review: https://git.soma.salesforce.com/ContentX/lwc/pull/929